src/pages/Environments.js

```diff
@@ -92,15 +92,20 @@ const DeleteDiskModal = ({ disk: { googleProject, name }, isGalaxyDisk, onDismis
   ])
 }
 
-const DeleteAppModal = ({ app: { googleProject, appName, diskName, appType }, onDismiss, onSuccess }) => {
+const DeleteAppModal = ({ app: { appName, diskName, appType, cloudContext: { cloudProvider, cloudResource } }, onDismiss, onSuccess }) => {
   const [deleteDisk, setDeleteDisk] = useState(false)
   const [deleting, setDeleting] = useState()
   const deleteApp = _.flow(
     Utils.withBusyState(setDeleting),
-    withErrorReporting('Error deleting cloud environment')
+    withErrorReportingInModal('Error deleting cloud environment', onDismiss)
   )(async () => {
-    await Ajax().Apps.app(googleProject, appName).delete(deleteDisk)
-    onSuccess()
+    //TODO: this should use types in IA-3824
+    if (cloudProvider === 'GCP') {
+      await Ajax().Apps.app(cloudResource, appName).delete(deleteDisk)
+      onSuccess()
+    } else {
+      throw new Error('unable to delete non-GCP apps currently - unimplemented')
```

Comment:
Nit: This error message could be a bit more user friendly. This looks like a temporary message mainly intended for developers, but there's still a chance it'll end up being shown to users at some point.

Maybe something like:
```suggestion
      throw new Error('Deleting apps is currently only supported on GCP')
```